pkg/api/norman/customization/workload/validator.go

```diff
@@ -0,0 +1,24 @@
+package workload
+
+import (
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/norman/types/convert"
+	"github.com/rancher/norman/types/values"
+)
+
+// Validates deprecated fields `environment` and `environmentFrom` are not being used. These fields were deprecated in favor
+// of the k8s native types `env` and `envFrom` for being unable to preserve the order of the variables and break interpolation.
+func Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
+	if containers, _ := values.GetSlice(data, "containers"); len(containers) > 0 {
+		for _, c := range containers {
+			if !convert.IsAPIObjectEmpty(values.GetValueN(c, "environment")) {
+				return httperror.NewAPIError(httperror.InvalidBodyContent, "field `environment` is deprecated, please use Kubernetes native types for the container's environment variables")
```

Comment:
could we update msg to say `please use Kubernetes native type env...`? and `envFrom` for L19